MTA-7135: Agent Mode Compilation Guardrails AST Verification (#380)
migtools/mta-documentation: docs/topics/developer-lightspeed/proc_enabling-agent-mode.adoc

// Module included in the following assemblies:
//
// * docs/topics/developer-lightspeed/assembly_understanding_dl.adoc

:_mod-docs-content-type: PROCEDURE
[id="enabling-agent-mode_{context}"]
= Enabling agent mode

[role="_abstract"]

> Enable agent mode in {mta-dl-plugin} through the VS Code command palette or the extension settings panel to verify code suggestions automatically with a local build dry run.

Reviewer: Just to comply with the "Why then what" template
```suggestion
To verify code suggestions automatically with a local build dry run, enable agent mode in {mta-dl-plugin} through the Visual Studio Code command palette or the extension settings panel.
```

.Prerequisites

> * You have installed the {mta-dl-plugin} extension in VS Code.

Reviewer: ```suggestion
* You have installed the {mta-dl-plugin} extension in Visual Studio Code.
```

> * Your project has a supported build file (the `pom.xml`, `build.gradle`, or `package.json` file) at the repository root.

Reviewer: ```suggestion
* Your project has a supported build file (`pom.xml`, `build.gradle`, or `package.json`) at the repository root.
```

> * The corresponding build tool (Maven, Gradle, or npm) is installed and available on your system `PATH`.

Reviewer: ```suggestion
* You have installed the corresponding build tool (Maven, Gradle, or npm) and added it to your system `PATH`.
```

.Procedure

. Enable agent mode in one of the following ways:
+
--
* By using the command palette:

.. Press Ctrl+Shift+P on Linux or Windows, or Cmd+Shift+P on macOS, to open the command palette.
.. Type `Preferences: Open User Settings (JSON)` to open the `settings.json` file.
.. Set the `mta-vscode-extension.genai.agentMode` parameter to `true`.

* By using the extension settings panel:

.. Click **Extensions** > **{mta-dl-plugin}** > **Settings**.
.. Select the **Agent Mode** checkbox to enable agent mode.

> --

Reviewer: This open block syntax (--) isn't needed. Should render just fine without it
```suggestion

* By using the command palette:

.. Open the command palette by pressing Ctrl+Shift+P on Linux or Windows, or Cmd+Shift+P on macOS.
.. Type `Preferences: Open User Settings (JSON)` to open the `settings.json` file.
.. Set the `mta-vscode-extension.genai.agentMode` parameter to `true`.

* By using the extension settings panel:

.. Click *Extensions* > *{mta-dl-plugin}* > *Settings*.
.. Enable agent mode by selecting the *Agent Mode* checkbox.
```

. Restart the IDE language server when you are prompted to apply the change.

.Verification

> . Open the Lightspeed output panel by clicking **View** > **Output** > **Red{nbsp}Hat Developer Lightspeed**, and request a code suggestion.

Reviewer: ```suggestion
. Open the Lightspeed output panel by clicking *View* > *Output* > *Red{nbsp}Hat Developer Lightspeed*.
. Request a code suggestion.
```
